### PR TITLE
Making Annoy resources closeable

### PR DESCRIPTION
**Pull request by eshvk**
@edwardnewett @yonromai

### CODE REVIEW

File: src/main/java/com/spotify/annoy/ANNIndex.java [hunk 1]
```diff
@@ -67,7 +68,7 @@ public ANNIndex(final int dimension,
 
 
   private void load(final String filename) throws IOException {
-    RandomAccessFile memoryMappedFile = new RandomAccessFile(filename, "r");
+    this.memoryMappedFile = new RandomAccessFile(filename, "r");
```
Comment (yonromai) [inline src/main/java/com/spotify/annoy/ANNIndex.java]: do you need this `this`? You're not using it in the rest of the function

File: src/main/java/com/spotify/annoy/ANNIndex.java [hunk 2]
```diff
@@ -140,6 +141,24 @@ public static float euclideanMargin(final float[] u,
...
+   * close may fail require careful attention. It is strongly advised
+   * to relinquish the underlying resources and to internally
+   * <em>mark</em> the {@code Closeable} as closed, prior to throwing
+   * the {@code IOException}.
+   *
+   * @throws IOException if an I/O error occurs
+   */
+  @Override
+  public void close() throws IOException {
+    this.memoryMappedFile.close();
```
Comment (yonromai) [inline src/main/java/com/spotify/annoy/ANNIndex.java]: same

**Comment by yonromai** [comment]
Awesome!

**Comment by eshvk** [comment]
@yonromai Good catch, fixed that.

**Comment by yonromai** [comment]
Boom!

**Comment by a1k0n** [comment]
Nice. With Closeable and everything.

**Pull request merged**